### ISSUE TITLE
Encrypted drives in OMV6 re-lock themselves

### ISSUE THREAD

**Issue by Koliera** [OpenMediaVault-Plugin-Developers/openmediavault-luksencryption#51]
Have encountered a weird bug where the drives (after some time of them being unlocked) just lock themselves and I need to go to the OMV GUI and unlock them again. Bit of a hassle and haven't seen anything in the notifications that might show why this is happening.

**Comment by ryecoaaron** [comment]
I assume you are using usb drives?  If yes, they are lock when the usb controller puts the drive to sleep which is like disconnecting the drive.  I don't know of a way to prevent this.

**Comment by Koliera** [comment]
Hi, tope, they're two hdds.
Though, I do use ProxMox and have passed thru the hdd to the omv vm. It used to work great before, but I've updated to both the latest proxmox and latest omv ( was running on older versions without issue for last 18 months).

**Comment by ryecoaaron** [comment]
Are you setting the drives to spin down?  OMV did just switch from hdparm to smartctl in 7.0.4-2
https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/debian/changelog#L9

**Comment by Koliera** [comment]
Everything looks to be disabled
![hdd](https://github.com/OpenMediaVault-Plugin-Developers/openmediavault-luksencryption/assets/78056083/3cfb7ffe-0fde-42a9-b60f-a57796964fc9)

**Comment by ryecoaaron** [comment]
I can't explain it then.  Something is causing a disconnect.  The plugin isn't a service and isn't constantly interacting with anything.  I would look for messages in dmesg.  It would be worth your time to post on the forum where people who actually use luks in this way might have run into this before.

**Comment by Koliera** [comment]
Interesting. In any case, would (as per the example picture) the command to decrypt the hdd be the following:
"cryptsetup-reencrypt --decrypt /dev/sdc"

**Comment by Koliera** [comment]
Yes, correct, that's to decrypt them once, I was thinking to completely remove the encryption on the hdd.

**Comment by ryecoaaron** [comment]
I've worked with LUKS for years and never tried that.  Seems like it would still sit on a LUKS device with is weird.  I will have to try it.

**Comment by Koliera** [comment]
This looks to be the one as of 2.6.0
cryptsetup reencrypt <device-name> --decrypt
Will try over the next day.

**Comment by ryecoaaron** [comment]
You are missing the --header flag.

I just decrypted a container on my test system.  It "moved" the ext4 filesystem from the container to the drive.  So, it seems like it works well.  Since fstab is mounting the filesystem, it just worked on reboot as well.

**Comment by Koliera** [comment]
Interesting, maybe the decryption option could be added to the plugin gui.
Also, noticed something interesting - I unencrypted it via the terminal using the luksOpen command and it worked and didn't re-crypt BUT for some reason it didn't actually mount the unencrypted drives in OMV.

**Comment by ryecoaaron** [comment]
Since these filesystems are in fstab, there is nothing actively watching them to mount when they get unlocked.

**Comment by Koliera** [comment]
Hmm, so why is it when I decrypt using the plugin they remount automatically? 
The default situation is that once the drives re-encrypt the mounted volumes show them as missing and once decrypted they automatically go online.

**Comment by ryecoaaron** [comment]
the plugin tries to run a mount command after decrypting.

**Comment by Koliera** [comment]
Excellent, that explains it. For the simple people like myself - what would be that mounting command that's run?
